pathfinderlib/main.py

```diff
@@ -2,22 +2,60 @@
 import os
 import tempfile
 import argparse
+import inspect
 import json
+import re
+import pandas
+from Bio import SeqIO
 
 from pathfinderlib.version import __version__
-#path to humann
 
-default_config ={
-    "humann_fp": "/home/ashwini/ash/other_softwares/humann/",
-    "kegg_fp": "/home/ashwini/ash/kegg/kegg"
-    }
-        
-                                
-class Humann(object):
-    def __init__(self, config):
-        self.config = config
+def get_config(user_config_file):
+    config ={
+        "humann_fp": "/home/ashwini/ash/other_softwares/humann/",
+        "kegg_fp": "/home/ashwini/ash/kegg/kegg",
+        "kegg_idx_fp":"",
+        "kegg_to_ko_fp":"",
+        "rap_search_fp": "",
+        "search_method":"rapsearch", # rapsearch or blastx
+        "mapping_method":"best_hit", # best_hit or humann
+        "evalue_cutoff":0.001,
+        "num_threads":4
+        }
+
+    if user_config_file is None:
+        default_user_config_fp = os.path.expanduser("~/.path_finder.json")
+        if os.path.exists(default_user_config_fp):
+            user_config_file = open(default_user_config_fp)
+
+    if user_config_file is not None:
+        user_config = json.load(user_config_file)
+        config.update(user_config)
+        return config
+
+def returnObject(tool_cls, config):
+    # Proceed stepwise here to improve quality of error messages.
+    tool_args = []
+    for argname in tool_cls.get_argnames():
+        arg = config[argname]
+        tool_args.append(arg)
+    return tool_cls(*tool_args)
+
+def Alignment(config):
+    tool_cls = search_methods_available[config["search_method"]]
+    return returnObject(tool_cls, config)
+
+class _Alignment(object):
+    def __init__(self, search_method, kegg_fp, num_threads):
+        self.search_method = search_method
+        self.kegg_fp = kegg_fp
+        self.num_threads = num_threads
 
-    def make_fastq_to_fasta_command(self, filename):
+    @classmethod
+    def get_argnames(cls):
+        return inspect.getargspec(cls.__init__)[0][1:]
+
+    def _make_fastq_to_fasta_command(self, filename):
         return [
             "seqtk",
```

Comment:
Should make note about seqtk dependency somewhere in readme.  May want to do this in python to eliminate a dependency, though I am okay with seqtk also because it's sooo fast.